core/src/main/java/org/mule/execution/AsyncResponseFlowProcessingPhase.java

```diff
@@ -98,6 +98,18 @@ private void fireNotification(MuleEvent event, int action)
     {
         try
         {
+            if (event == null)
```

Comment:
Why does it have to be the last event? I think with the transport in this scenario it used the request event (this may have been incorrect, but it's worth comparing with transport).